include/input_iterator.hpp

```diff
@@ -1,25 +1,78 @@
+// ============================= INPUT ITERATOR ============================= //
+// Project: The Experimental Bit Algorithms Library
+// Name: input_iterator.hpp
+// Description: input_iterator is a wrapper for normal iterators which constrains
+// the how we can use the wrapped iterator to match that of an InputIterator 
+// Creator: Vincent Reverdy
+// Contributor(s): Collin Gress [2019] 
+// License: BSD 3-Clause License
+// ========================================================================== //
+#ifndef _INPUT_ITERATOR_HPP_INCLUDED
+#define _INPUT_ITERATOR_HPP_INCLUDED
+// ========================================================================== //
+
+// ============================== PREAMBLE ================================== //
+// C++ standard library
+// Project sources
+// Third-party libraries
+// Miscellaneous
```

Comment:
<iostream> should be under C++ standard library, and the preamble needs to be closed with no newlines. We also need the other separators as well as in vincents bit_iterator